src/material/dialog/_mdc-dialog-structure-overrides.scss

```diff
@@ -38,19 +38,18 @@
       max-width: inherit;
     }
 
-    // MDC by default sets the `surface` to `display: flex`. MDC does this in order to
-    // be able to make the content scrollable while locking the title and actions. This
-    // does not work in our dialog anyway because due to the content projection, arbitrary
-    // components can be immediate children of the surface and make this a noop. If only
-    // templates or DOM elements are projected, the flex display could cause unexpected
-    // alignment issues as explained in our coding standards (see `CODING_STANDARDS.md`).
-    // Additionally, the surface by default should expand based on the parent overlay
-    // boundaries (so that boundaries for the overlay config are respected). The surface
-    // by default would only expand based on its content.
     .mdc-dialog__surface {
-      display: block;
       width: 100%;
       height: 100%;
     }
+
+    // MDC sets the `surface` to `display: flex`. MDC does this in order to be able to make the
+    // content scrollable while locking the title and actions. This only works in our dialog if we
+    // set the children of the `surface` to `display: contents`. This is because, due to content
+    // projection, arbitrary components can be immediate children of the surface and make this a
+    // noop.
+    .mat-mdc-dialog-surface > * {
+      display: contents;
```

Comment:
It looks like `display: contents` has an effect over a11y: https://developer.mozilla.org/en-US/docs/Web/CSS/display#display_contents. Couldn't we use some other value?

Reply:
Shouldn't this be ok since browsers will only remove the elements with `display: contents` but keep the descendants? From what I can tell, it shouldn't impact accessibility if the host element is removed since the direct children of `surface` is just the host element of the component passed into `dialog.open`

Reply:
Sure, but `.mat-mdc-dialog-surface` is the wrapper directly around the user's content. For example, if a `button` was projected directly into the dialog `.mat-mdc-dialog-surface > *` would make it `display: contents` and hide it from assistive technology.

Reply:
Generally `display: contents` does not remove dom node from the a11y tree. That's at least form what I've seen, but accessibility trees are not consistent across browsers. 

If the a11y tree looks the same on Chrome and Firefox, then I don't think this would break anything. That said,  I don't know what all the possibilities for children of `.mat-mdc-dialog-surface`. I think it would be safer to use something else. I think Kristiyan is right that  display: contents might cause problems that we don't anticipate.

I don't feel super strongly against it, but `display: contents`  would not be my first choice here. 

> MDC sets the `surface` to `display: flex`

How hard is it to use something other than `display: contents`? We have the ability to override styles.

Reply:
While it's not according to the spec, MDN mentions that some browsers will remove the `display: contents` element from the a11y tree:

> Current implementations in some browsers will remove from the [accessibility tree](https://developer.mozilla.org/en-US/docs/Learn/Accessibility/What_is_accessibility#accessibility_apis) any element with a display value of contents (but descendants will remain).


Another issue that this change can cause is that it might cause elements with `display: none` to be shown.

Reply:
> Sure, but `.mat-mdc-dialog-surface` is the wrapper directly around the user's content. For example, if a `button` was projected directly into the dialog `.mat-mdc-dialog-surface > *` would make it `display: contents` and hide it from assistive technology.

I'm skeptical that this is a common enough use case that should block this from landing.

> Another problem is that this could reset elements with `display: none`.

This will actually interfere with basically all apps using the dialog. This change will only be feasible if we also release a schematic to add a mixin to override this change back to the current styles.

Reply:
I don't follow the argument about display: none interfering with the dialog. Currently all the elements inside TemplateRef-based dialogs are projected directly inside the dialog surface (see the dialog demo in the dev app). This change would target and override the display value of all of these elements.